### PR TITLE
Refactor and Improve error handling code of SourceHandler

### PR DESCRIPTION
**Pull request by shafreenAnfar**

### CODE REVIEW

File: components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/listener/SourceHandlerErrorHandler.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+            handleIncompleteInboundRequest(Constants.REMOTE_CLIENT_ABRUPTLY_CLOSE_CONNECTION);
+        }
+    }
+
+    void handleIdleErrorScenario() {
+        try {
+            switch (state) {
+            case CONNECTED:
+                serverConnectorFuture.notifyErrorListener(
+                        new ServerConnectorException(Constants.IDLE_TIMEOUT_TRIGGERED_BEFORE_READING_INBOUND_RESPONSE));
```
Comment (chamil321) [inline components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/listener/SourceHandlerErrorHandler.java]: INBOUND_REQUEST*

File: components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/listener/SourceHandlerErrorHandler.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+                // after that.
+                serverConnectorFuture.notifyErrorListener(new ServerConnectorException(
+                        Constants.IDLE_TIMEOUT_TRIGGERED_BEFORE_WRITING_OUTBOUND_RESPONSE));
+
+                // This is to stop application layer writing response after
+                // idle timeout is triggered.
+                httpOutboundRespFuture.notifyHttpListener(new ServerConnectorException(
+                        Constants.IDLE_TIMEOUT_TRIGGERED_BEFORE_WRITING_OUTBOUND_RESPONSE));
+                break;
+            default:
```
Comment (chamil321) [inline components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/listener/SourceHandlerErrorHandler.java]: Shouldn't we consider setStateHeaderReceived specifically?

File: components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/listener/SourceHandlerErrorHandler.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+            handleIncompleteInboundRequest(Constants.EXCEPTION_CAUGHT_WHILE_READING_REQUEST);
+        }
+        try {
+            serverConnectorFuture.notifyErrorListener(cause);
+        } catch (ServerConnectorException e) {
+            log.error("Couldn't notify error state to application layer");
+        }
+    }
+
+    public void setStateConnected() {
```
Comment (chamil321) [inline components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/listener/SourceHandlerErrorHandler.java]: Shall we use single method call setState(InboundState state) which has the InboundState enum as parameter rather using separate methods for each state?

**Comment by shafreenAnfar** [comment]
Closing this PR as there is another one to address same PR.

https://github.com/wso2/transport-http/pull/179

**Pull request closed without merging**